direnvrc

```diff
@@ -31,28 +40,38 @@ _nix() {
   nix --extra-experimental-features "nix-command flakes" "$@"
 }
 
+_require_version() {
+  local cmd=$1 version=$2 required=$3
+  if ! printf "%s\n" "$required" "$version" | sort --check=quiet --version-sort; then
```

Comment:
Does this check works with the macos sort? https://ss64.com/osx/sort.html

Reply:
Otherwise we could also try to detect if the bash version supports regex instead?

Reply:
Otherwise here is a pure bash version:

```
version_compare () {
    if [[ $1 == $2 ]]
    then
        return 0
    fi
    local IFS=.
    local i ver1=($1) ver2=($2)
    # fill empty fields in ver1 with zeros
    for ((i=${#ver1[@]}; i<${#ver2[@]}; i++))
    do
        ver1[i]=0
    done
    for ((i=0; i<${#ver1[@]}; i++))
    do
        if [[ -z ${ver2[i]} ]]
        then
            # fill empty fields in ver2 with zeros
            ver2[i]=0
        fi
        if ((10#${ver1[i]} > 10#${ver2[i]}))
        then
            return 1
        fi
        if ((10#${ver1[i]} < 10#${ver2[i]}))
        then
            return 2
        fi
    done
    return 0
}
```

Reply:
Verified on macOS 14.1.2

Reply:
Ok. Just to double check, you were using the builtin sort executable and not sort from coreutils here?

Reply:
```
arthur@Mac-mini nix-direnv % command -v sort
/usr/bin/sort
arthur@Mac-mini nix-direnv % sort --version
2.3-Apple (165.0.4)
```